### PR TITLE
test(api): add comprehensive auth middleware test coverage

### PR DESCRIPTION
**Pull request by andrewxhill**
## Summary
- Add 23 comprehensive test cases for authentication middleware
- Achieve 100% coverage on critical security boundary (189 lines, previously 0%)
- Cover all authentication methods: Privy JWT, Agent API keys, Admin API keys
- Test authentication fallback chain and error handling
- Validate request context population for all auth paths

## Why This Area Was Chosen

| Criterion | Score (1-5) | Notes |
|-----------|-------------|-------|
| **Blast Radius** | 5 | Every API request flows through this code |
| **Bug Likelihood** | 5 | Complex multi-path logic with 3 auth methods |
| **Observability** | 3 | Logs present, but auth failures can be silent |
| **Criticality** | 5 | Security boundary for entire API |
| **Single File** | ✅ | Isolated testing with mocked dependencies |
| **Total Score** | **18/20** | Highest priority untested area |

### New Tests (23 total)

**Privy JWT Authentication (5 tests)**
- Valid token with existing user → Sets req.userId, calls next()
- Valid token for login endpoint without user → Backward compatibility
- Valid token but user not found (non-login) → Returns 401
- Invalid token → Falls through to API key auth
- No token → Falls through to API key auth

**Agent API Key Authentication (5 tests)**
- Valid agent API key → Sets req.agentId and req.userId
- Agent exists but owner lookup fails → Sets agentId only
- Invalid agent API key → Falls through to admin auth
- Agent validation throws error → Falls through to admin auth
- No API key → Returns 401

**Admin API Key Authentication (6 tests)**
- Valid admin API key → Sets req.adminId, req.isAdmin, req.admin
- Username fallback when name not available
- AdminId set even when admin lookup fails
- Invalid admin API key → Returns 401
- Admin validation throws error → Returns 401

**Edge Cases and Error Handling (5 tests)**
- No auth credentials → Returns 401
- All methods fail → Returns 401 with appropriate message
- Unexpected errors → Handled gracefully
- Sensitive info not exposed in errors
- Authentication fallback chain tested

**Authentication Fallback Chain (2 tests)**
- Privy → Agent → Admin sequence works correctly
- Stops at first successful authentication

## Coverage Impact

**Before:** Authentication middleware had 0% test coverage (189 lines untested)  
**After:** 100% coverage via 23 comprehensive tests

## Test Plan

- [x] All new tests pass (23/23)
- [x] TypeScript compiles with no errors
- [x] ESLint passes with no warnings
- [x] Prettier formatting verified
- [x] Existing tests still pass
- [x] Authentication paths correctly populate request context
- [x] Fallback logic preserves security (no bypasses on errors)
- [x] Error handling doesn't expose sensitive information

## Risk Reduction

This test suite immediately protects against:
- ✅ Authentication bypass vulnerabilities
- ✅ Incorrect fallback logic allowing unauthorized access
- ✅ Request context population bugs
- ✅ Sensitive data leakage in error messages
- ✅ Regressions in security-critical code paths

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/api/src/middleware/__tests__/auth.middleware.test.ts [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+  let mockNext: ReturnType<typeof vi.fn>;
+
+  beforeEach(async () => {
+    vi.clearAllMocks();
+
+    // Create mock services
+    mockAgentService = {
+      validateApiKey: vi.fn(),
+      getAgent: vi.fn(),
+    } as unknown as MockedObject<AgentService>;
```
Comment (carsonfarmer) [inline apps/api/src/middleware/__tests__/auth.middleware.test.ts]: To avoid this mess (that the AIs seem to love to create), we're finding it super useful to use this package https://github.com/eratio08/vitest-mock-extended

@asutula used it here https://github.com/recallnet/js-recall/blob/main/apps/api/src/services/__tests__/boost-award.service.test.ts to really good effect, and it makes mocking type safe, none of the `as any` or `as unknown as X` nonsense.

**Review by carsonfarmer — CHANGES_REQUESTED** [review]
This is a very useful PR, let's make sure we get this in. One "standard" we're trying to create across the repos is to leverage the https://github.com/eratio08/vitest-mock-extended library to make mocking properly typed. Can you see how easy it is to get the agent to adopt this pattern?

File: apps/api/src/middleware/__tests__/auth.middleware.test.ts (hunk 1 above)
Comment (andrewxhill) [reply apps/api/src/middleware/__tests__/auth.middleware.test.ts]: done

File: apps/api/src/middleware/__tests__/auth.middleware.test.ts [hunk 2]
```diff
@@ -0,0 +1,578 @@
+import { Request, Response } from "express";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+import { type DeepMockProxy, mockDeep } from "vitest-mock-extended";
```
Comment (carsonfarmer) [inline apps/api/src/middleware/__tests__/auth.middleware.test.ts:3]: Nice

File: apps/api/src/middleware/__tests__/auth.middleware.test.ts [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+      vi.mocked(isLoginEndpoint).mockReturnValue(true);
+
+      await middleware(mockReq as Request, mockRes as Response, mockNext);
+
+      expect(mockNext).toHaveBeenCalledWith();
+      expect(mockReq.userId).toBeUndefined();
+    });
+
+    it("should reject non-login endpoint access for valid token without user", async () => {
+      const { extractPrivyIdentityToken } = await import(
```
Comment (carsonfarmer) [inline apps/api/src/middleware/__tests__/auth.middleware.test.ts:133]: Are these dynamic imports just because we mock them above?

**Review by carsonfarmer — APPROVED** [review]
Cheers for the quick turnaround!
